Update multus deployment, configuration, and verification

clacroix12 opened the pull request.

clacroix12 commented:
I've deployed successfully as well as ran the deployment verification successfully against this cluster.

Deployment: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/7729/
Verification: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/23853/

This is working for one scenario in which we configure both the public and cluster networks.

clacroix12 commented:
There are some additional steps needed to complete verification, so moving this back to WIP

ebenahar commented on conf/deployment/vsphere/upi_1az_rhcos_multus_vsan_3m_3w.yaml:
```diff
@@ -13,6 +13,8 @@ ENV_DATA:
   compute_memory: '65536'
   fio_storageutilization_min_mbps: 10.0
   is_multus_enabled: true
+  multus_create_public_net: true
+  multus_create_cluster_net: true
```
is the plan to create configuration files for covering more permutations of public net and cluster net?

clacroix12 replied:
Yes, should have those configs in today.

ebenahar commented on ocs_ci/deployment/deployment.py:
```diff
@@ -652,22 +652,74 @@ def deploy_ocs_via_operator(self, image=None):
         run_cmd(f"oc create -f {constants.OLM_YAML}")
 
-        # create multus network
+        # Create Multus Networks
```
are we doing the same also for nodes added post deployment?

clacroix12 replied:
I haven't added anything regarding nodes added post-deployment. I suspect we would want to enable PROMISC on the network interface for the newly added node. Are we running add_node tests in vsphere environments?

ebenahar replied:
Yes, add node tests are done on vSphere (I believe both UPI and IPI).
@clacroix12 @OdedViner should we handle the adjustments needed for Multus in add node functionality with this PR or a follow up one?

clacroix12 replied:
@ebenahar missed this update earlier but yes I think any changes necessary for test cases to work with multus deployments should be done in isolated PRs.

clacroix12 commented:
Added the ability to create either the public or cluster network individually as well as both (default behavior). I've also added configurations that should cover more of the deployment scenario permutations as described in our Multus deployment testing document: https://url.corp.redhat.com/6d49eae

OdedViner commented on conf/deployment/baremetal/upi_1az_rhcos_multus_cluster_nvme_intel_3m_3w copy.yaml:
@clacroix12 Can you fix the conf file name
`conf/deployment/baremetal/upi_1az_rhcos_multus_cluster_nvme_intel_3m_3w copy.yaml `->
`conf/deployment/baremetal/upi_1az_rhcos_multus_cluster_nvme_intel_3m_3w.yaml`

clacroix12 replied:
Fixed

clacroix12 commented on conf/deployment/vsphere/upi_1az_rhcos_multus_cluster_vsan_3m_3w .yaml:
```diff
@@ -0,0 +1,22 @@
+---
+# This config is suppose to work on most of DCs we have.
```
I had already fixed this, not sure how it was changed back, possibly due to rebasing. This should be fixed now.
